### ISSUE TITLE
Support non-canonical taxonomic ranks in the tree hierarchy

### ISSUE THREAD

**Issue by bruvellu** [nelas/cifonauta#305]
The taxon_update routine that fetches taxonomic information from the WoRMS database, only fetches and creates the canonical taxonomic ranks: Kingdom, Phylum, Class, Order, Family, Genus, Species. If someone adds a different rank like subphylum, infraclass or subspecies, they will not be properly integrated in the tree.

For example, adding Crustacea which is classified as a subphylum on WoRMS, does not link with its immediate descendants, such as the classes Malacostraca or Ostracoda:

![crustacea](https://github.com/bruvellu/cifonauta/assets/361179/f54db7ef-1253-4064-9e55-b6b7ea49406f)

Importing the entire WoRMS database to have a full taxonomic record is one possibility, but that brings a high maintenance burden and complicates everything due to its large size. Something I want to avoid.

Another solution is checking during the taxon_update if the immediate ancestor exists in the Cifonauta database. If it exists, the current taxon is linked to it; if not, it links to the next canonical rank. This way we can add key taxa and still keep the ancestor/descendant relationships in order.

The tree will probably become unbalanced, but that should be fixed in another issue.

**Comment by bruvellu** [comment]
This was a lot of work to implement (done in 02d4610c9b925a1ec1dbd594c6dbb3c8ba3774ef). While the logic for fetching the lineage up the tree was straightforward, there was a lot of refactoring and optimization of the code for updating the taxonomic records. Adding more taxa increases the complexity of the database and edge cases begin to appear. I had to make sure they were being handled properly. There's one case of two records with the same name, Cidaroida as a [subclass](https://marinespecies.org/aphia.php?p=taxdetails&id=510498) and [superfamily](https://marinespecies.org/aphia.php?p=taxdetails&id=852318), which I suspect is an error in the WoRMS classification (I notified them). Finally, I've also updated the management command that updates the taxonomic information. Now it is multiple times faster because it is caching the records fetched from WoRMS.
